docs/book/user-guide/advanced-guide/artifact-management/model-management.md

```diff
@@ -0,0 +1,267 @@
+```
+
+You can view some of the options of what can be passed into this command by
+running `zenml model register --help` but since you are using the CLI outside a
+pipeline run the arguments you can pass in are limited to non-runtime items. You
+can also associate tags with models at this point, for example, using the
+`--tag` option.
+
+### Explicit dashboard registration
+
+The Model Control Plane interface on the dashboard is currently in testing
+and will be released soon to ZenML Cloud users. Documentation for this feature
+will be available [here](./model-control-plane-dashboard.md) once it is released.
+
+### Explicit Python SDK registration
+
+You can register a model using the Python SDK as follows:
+
+```python
+from zenml.models import ModelVersion
+from zenml import Client
+
+Client().create_model(
+    name="iris_logistic_regression",
+    license="Copyright (c) ZenML GmbH 2023",
+    description="Logistic regression model trained on the Iris dataset.",
+    tags=["regression", "sklearn", "iris"],
+)
+```
+
+### Implicit registration by ZenML
+
+The most common use case for registering models is to do so implicitly as part
+of a pipeline run. This is done by specifying a `ModelVersion` object as part of
+the `model_version` argument of the `@pipeline` decorator.
+
+As an example, here we have a training pipeline which orchestrates the training
+of a model object, storing datasets and the model object itself as links within
+a newly created Model Version. This integration is achieved by configuring the
+pipeline within a Model Context using `ModelVersion`. The name and
+`create_new_model_version` fields are specified, while other fields remain optional for this task.
+
+```python
+from zenml import pipeline
+from zenml.model import ModelVersion
+
+@pipeline(
+    enable_cache=False,
+    model_config=ModelVersion(
+        name="demo",
+        license="Apache",
+        description="Show case Model Control Plane.",
+        create_new_model_version=True,
+        delete_new_version_on_failure=True,
+    ),
+)
+def train_and_promote_model():
+    ...
+```
+
+Running the training pipeline creates a model and a Model Version, all while
+maintaining a connection to the artifacts.
+
+## Model versions
+
+Each model can have many model versions. Model versions are a way for you to
+track different iterations of your training process, complete with some extra
+dashboard and API functionality to support the full ML lifecycle.
+
+### When are model versions created?
+
+Model versions are created implicitly as you are running your machine learning
+training, so you don't have to immediately think about this. If you want more
+control over how and when these versions are controlled, our API has you
+covered, with options for whether new model versions are created as well as for
+the deletion of new model versions when pipeline runs fail. [See
+above](model-management.md#explicit-python-sdk-registration) for how to create
```

Comment:
This ref doesn't show how to work with `version` param. We need to extend with it, e.g. `version="1.0.0"` for specific version naming would be a good start. Also we need to demo the difference between `version=None`(or skipped) and specific name to illustrate how model versions are numbered automatically.